packages/subscription-manager/src/components/UserSettings/UserSettings.tsx

```diff
@@ -17,36 +23,90 @@ type SubscriptionUserSettings = Partial< {
 
 type UserSettingsProps = {
 	value?: SubscriptionUserSettings;
-	onChange?: ( value: SubscriptionUserSettings ) => void;
-	loading?: boolean;
+	loading: boolean;
+};
+
+type NoticeProps = {
+	type: 'success' | 'warning' | 'error';
+	message: string;
 };
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars -- until we start using any of these props
-const UserSettings = ( { value = {}, loading = false, onChange }: UserSettingsProps ) => (
-	<div className="user-settings">
-		<EmailFormatInput
-			value={ value.mail_option ?? 'html' }
-			onChange={ ( evt: FormEvent< HTMLSelectElement > ) =>
-				onChange?.( { mail_option: evt.currentTarget.value as EmailFormatType } )
-			}
-		/>
-		<DeliveryWindowInput
-			dayValue={ value.delivery_day ?? 0 }
-			hourValue={ value.delivery_hour ?? 0 }
-			onDayChange={ ( evt: FormEvent< HTMLSelectElement > ) =>
-				onChange?.( { delivery_day: parseInt( evt.currentTarget.value ) as DeliveryWindowDayType } )
-			}
-			onHourChange={ ( evt: FormEvent< HTMLSelectElement > ) =>
-				onChange?.( {
-					delivery_hour: parseInt( evt.currentTarget.value ) as DeliveryWindowHourType,
-				} )
-			}
-		/>
-		<BlockEmailsSetting
-			value={ value.blocked ?? false }
-			onChange={ ( value ) => onChange?.( { blocked: !! value.target.value } ) }
-		/>
-	</div>
-);
+const UserSettings = ( { value = {}, loading }: UserSettingsProps ) => {
+	const [ formState, setFormState ] = useState< SubscriptionUserSettings >( value );
+	const { mutate, isLoading, isSuccess, error } =
+		Reader.useSubscriptionManagerUserSettingsMutation();
+	const [ notice, setNotice ] = useState< NoticeProps | null >( null );
+
+	useEffect( () => {
+		// check if formState is empty object
+		if ( value ) {
+			setFormState( value );
+		}
+	}, [ value ] );
+
+	useEffect( () => {
+		if ( isSuccess ) {
+			setNotice( { type: 'success', message: translate( 'Settings saved' ) as string } );
+		}
+		if ( error ) {
+			setNotice( { type: 'error', message: error.message } );
+		}
```

Comment:
True. But since the type prop is also type checked (`type?: 'success' | 'warning' | 'error';`), exporting type constants will increase bundle size, while we get validation / prop checking for free if we  leave it to TypeScript.